src/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumeration.java

```diff
@@ -545,6 +554,18 @@ public String longestCommonSubsequence (String a, String b) {
 		return hirschberg.getLCS(a, b);
 	}
 
+	private boolean shouldContinue(List<Context> contextList) {
+		Iterator<Context> contextIterator = contextList.iterator();
+		boolean hasAuth = false;
+		while (contextIterator.hasNext()) {
+			Context aContext = contextIterator.next();
+			if(aContext.getAuthenticationMethod().getType().getName().equals("Form-based Authentication")) {
```

Comment:
Unfortunately there doesn't seem to be an enum or constant associated with AuthenticationMethodType(s).